dgv/impact/task_functions.py

```diff
@@ -0,0 +1,188 @@
+                    dataset_supervisors = [m['user']['id'] for m in dataset_supervisors['members']]
+                else:
+                    dataset_supervisors = [dataset['owner']['id']] if dataset['owner'] else []
+                legit = datagouv_team + dataset_supervisors
+
+                # getting time to legit response
+                opening_date = discussion['discussion'][0]['posted_on'][:10]
+                answered_date = None
+                for comment in discussion['discussion'][1:]:
+                    if comment['posted_by']['id'] in legit:
+                        answered_date = comment['posted_on'][:10]
+                        nb_discussions_with_legit_answer += 1
+                if not answered_date:
+                    time_to_answer.append(30)
+                else:
+                    opening_date = datetime.strptime(opening_date, "%Y-%m-%d")
+                    answered_date = datetime.strptime(answered_date, "%Y-%m-%d")
+                    delai = answered_date - opening_date
+                    time_to_answer.append(min(delai.days, 30))
+            else:
+                time_to_answer.append(30)
+    average_time_to_answer = round(np.mean(time_to_answer), 2)
+    print(f"Taux de réponses légitimes : {round(nb_discussions_with_legit_answer/nb_discussions*100)}%")
+
+    data = [
+        {
+            'administration_rattachement': 'DINUM',
+            'nom_service_public_numerique': 'data.gouv.fr',
+            'indicateur': 'Score qualité moyen 1000 JdD les plus vus',
+            'valeur': average_quality_score,
+            'unite_mesure': '%',
+            'est_cible': False,
+            'frequence_monitoring': 'mensuelle',
+            'date': datetime.today().strftime("%Y-%m-%d"),
+            'est_periode': False,
+            'date_debut': '',
+            'est_automatise': True,
+            'source_collecte': 'script',
+            'code_insee': '',
+            'dataviz_wish': 'barchart',
+            'commentaires': ''
+        },
+        {
+            'administration_rattachement': 'DINUM',
+            'nom_service_public_numerique': 'data.gouv.fr',
+            'indicateur': 'Délai moyen pour une réponse légitime à une discussion',
+            'valeur': average_time_to_answer,
+            'unite_mesure': 'jour',
+            'est_cible': False,
+            'frequence_monitoring': 'mensuelle',
+            'date': end_date,
+            'est_periode': True,
+            'date_debut': start_date,
+            'est_automatise': True,
+            'source_collecte': 'script',
+            'code_insee': '',
+            'dataviz_wish': 'barchart',
+            'commentaires': 'les délais sont écrétés à 30 jours'
+        },
+    ]
+    df = pd.DataFrame(data)
+    history = pd.read_csv(f'{DATADIR}/history.csv')
+    final = pd.concat([df, history])
+    df.to_csv(os.path.join(DATADIR, "statistiques_impact_datagouvfr.csv"), index=False, encoding="utf8")
+
+
+def send_stats_to_minio():
```

Comment:
can we duplicate saving file, one with this name, one with this name and the actual date. To keep an history in case there is some problem ?

Reply:
Done with last commit ✅